recipes/llm-voice-assistant/dotnet/LLMVoiceAssistant/Program.cs

```diff
@@ -0,0 +1,751 @@
+        }
+
+        private class Listener
+        {
+            readonly Cheetah cheetah;
+            readonly Porcupine porcupine;
+            readonly PvRecorder recorder;
+            event EventHandler<short[]>? AudioReceived;
+            string transcript = "";
+            public event EventHandler<string>? UserInputReceived;
+            public event EventHandler? WakeWordDetected;
+
+            public Listener()
+            {
+                cheetah = Cheetah.Create(accessKey: config.AccessKey,
+                                   endpointDurationSec: config.CheetahEndpointDurationSec,
+                                   enableAutomaticPunctuation: true);
+                porcupine =
+                    config.KeywordModelPath == null
+                        ? Porcupine.FromBuiltInKeywords(
+                              accessKey: config.AccessKey,
+                              keywords: [BuiltInKeyword.PICOVOICE],
+                              sensitivities: [config.PorcupineSensitivity])
+                        : Porcupine.FromKeywordPaths(
+                              accessKey: config.AccessKey,
+                              keywordPaths: [config.KeywordModelPath],
+                              sensitivities: [config.PorcupineSensitivity]);
+                recorder = PvRecorder.Create(frameLength: porcupine.FrameLength);
+                cheetahProfiler?.Init(cheetah.SampleRate);
+                porcupineProfiler?.Init(porcupine.SampleRate);
+                AudioReceived += PorcupineOnAudioReceived;
+                WakeWordDetected += OnWakeWordDetected;
+                UserInputReceived += OnUserInputReceived;
+            }
+
+            public void Run()
+            {
+                recorder.Start();
+                while (true)
+                {
+                    short[] pcm = recorder.Read();
+                    AudioReceived?.Invoke(this, pcm);
+                }
+            }
+
+            private void PorcupineOnAudioReceived(object? sender, short[] pcm)
+            {
+                porcupineProfiler?.Tick();
+                int keywordIndex = porcupine.Process(pcm);
+                porcupineProfiler?.Tock(pcm);
+                if (keywordIndex >= 0)
+                {
+                    WakeWordDetected?.Invoke(this, EventArgs.Empty);
+                    llmFinish.Task.Wait();
+                    Console.WriteLine("\nWake word detected, utter your request or question ...");
+                    Console.Write("User > ");
+                }
+            }
+
+            private void CheetahOnAudioReceived(object? sender, short[] pcm)
+            {
+                cheetahProfiler?.Tick();
+                CheetahTranscript transcript = cheetah.Process(pcm);
+                cheetahProfiler?.Tock(pcm);
+                if (transcript.Transcript.Length > 0)
+                {
+                    this.transcript += transcript.Transcript;

```

Comment:
having to have `transcript` and `this.transcript` in the same block is a good indication that you should change one of them to reduce confusion. I recommend changing the local var `transcript` to `result` or something